library/src/main/java/com/novoda/downloadmanager/lib/DownloadExecutorFactory.java

```diff
@@ -1,24 +1,28 @@
 package com.novoda.downloadmanager.lib;
 
 import android.content.Context;
-import android.content.pm.ApplicationInfo;
-import android.content.pm.PackageManager;
-import android.os.Bundle;
 
-import com.novoda.notils.logger.simple.Log;
-
-import java.util.concurrent.ExecutorService;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
 class DownloadExecutorFactory {
+
     private static final int KEEP_ALIVE_TIME = 10;
-    private static final int DEFAULT_MAX_CONCURRENT_DOWNLOADS = 5;
-    private static final String METADATA_MAX_CONCURRENT_DOWNLOADS = "com.novoda.downloadmanager.MaxConcurrentDownloads";
 
-    public ExecutorService createExecutor(Context context) {
-        int maxConcurrentDownloads = getMaximumConcurrentDownloads(context);
+    private final int maxConcurrentDownloads;
+
+    static DownloadExecutorFactory newInstance(Context context) {
+        ConcurrentDownloadsLimitProvider concurrentDownloadsLimitProvider = ConcurrentDownloadsLimitProvider.newInstance(context);
+        int maxConcurrentDownloads = concurrentDownloadsLimitProvider.getConcurrentDownloadsLimit();
```

Comment:
i feel the "cleanliness" needs to look horrible so that next person can see what we are doing, hiding the unnecceessary abstraction (and keeping more new's in one place)